fastlane/lib/fastlane/actions/get_github_release.rb

```diff
@@ -7,43 +7,38 @@ module SharedValues
     class GetGithubReleaseAction < Action
       def self.run(params)
         UI.message("Getting release on GitHub (#{params[:server_url]}/#{params[:url]}: #{params[:version]})")
-        require 'excon'
-        require 'base64'
 
-        server_url = params[:server_url]
-        server_url = server_url[0..-2] if server_url.end_with? '/'
-
-        headers = { 'User-Agent' => 'fastlane-get_github_release' }
-        headers['Authorization'] = "Basic #{Base64.strict_encode64(params[:api_token])}" if params[:api_token]
-
-        # To still get the data when a repo has been moved
-        middlewares = Excon.defaults[:middlewares] + [Excon::Middleware::RedirectFollower]
-        response = Excon.get("#{server_url}/repos/#{params[:url]}/releases", headers: headers, middlewares: middlewares)
-
-        case response[:status]
-        when 404
-          UI.error("Repository #{params[:url]} cannot be found, please double check its name and that you provided a valid API token (if it's a private repository).")
-          return nil
-        when 401
-          UI.error("You are not authorized to access #{params[:url]}, please make sure you provided a valid API token.")
-          return nil
-        else
-          if response[:status] != 200
-            UI.error("GitHub responded with #{response[:status]}:#{response[:body]}")
-            return nil
+        GithubApiAction.run(
+          server_url: params[:server_url],
+          api_token: params[:api_token],
+          http_method: 'GET',
+          path: "repos/#{params[:url]}/releases",
+          errors: {
+            404 => proc do |result|
+              UI.error("Repository #{params[:url]} cannot be found, please double check its name and that you provided a valid API token (if it's a private repository).")
+              return nil
+            end,
+            401 => proc do |result|
```

Comment:
Yeah - I agree - my thinking here was to ensure no change to existing setups _at all_. So anyone that is relying on this error message, will continue to see this error message. Perhaps users are regexing for this message before trying something new (i have no idea).

Happy to change it, but the aim of this PR was always to ensure every existing setup would continue to work. Let me know if you'd like me to change it to the base github api action. For instance, the base github api action catch all for all errors will display much more information.